From 7d15888d1fad998df059b809f5ecddaf9434c764 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 28 Jul 2020 15:28:13 +0300
Subject: [PATCH] [SW-12754] preboot: security: notify ready

LKD must not try to access the device memory space until the security
settings are not applied.
Add a notification to LKD that the security settings were applied and
device can be accessed now.
Currently this is a dummy, so we can integrate with LKD.
Later the security code will be added before this notification.

Change-Id: I18c45bafe9963475faeea58a6dd8b178f70c4d6b
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
(cherry picked from commit 6ce57a8df093bf69e3e14e364d1a9e1b9849c3d2)
---

diff --git a/preboot/preboot.c b/preboot/preboot.c
index 01875fe..881f3ad 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -160,6 +160,8 @@
 	/* Enable the errors communication */
 	cpu_boot_err_clrset(CPU_BOOT_ERR0_ENABLED, true);
 
+	cpu_boot_sts_set(CPU_BOOT_STATUS_SECURITY_READY);
+
 	/*
 	 * Note: we assume CPU frequency is not changed while CPU is running,
 	 * so the udelay init is done once before we start.
